pages/services/kafka-zookeeper/2.2.0-3.4.11/security/index.md

```diff
@@ -0,0 +1,295 @@
+```json
+{
+    "service": {
+        "security": {
+            "kerberos": {
+                "advanced": {
+                    "required_for_quorum_learner": true,
+                    "required_for_quorum_server": true,
+                    "required_for_client": false
+                }
+            }
+        }
+    }
+}
+```
+and deploy this as a configuration update:
+```bash
+$ dcos kafka-zookeeper --name=<service name> update start --options=kerberos-toggle-step-3.json
+$ dcos kafka-zookeeper --name=<service name> plan show deploy
+deploy (serial strategy) (COMPLETE)
+└─ node-update (serial strategy) (COMPLETE)
+   ├─ zookeeper-0:[server, metrics] (COMPLETE)
+   ├─ zookeeper-1:[server, metrics] (COMPLETE)
+   └─ zookeeper-2:[server, metrics] (COMPLETE)
+```
+{{ model.techName }} will now require Kerberos authentication for the entire leader election process.
+
+The final step is to require Kerberos authentication for clients connecting to the DC/OS {{ model.techName }} instance with an options file (say `kerberos-toggle-step-4.json`) as follows:
+```json
+{
+    "service": {
+        "security": {
+            "kerberos": {
+                "advanced": {
+                    "required_for_quorum_learner": true,
+                    "required_for_quorum_server": true,
+                    "required_for_client": true
+                }
+            }
+        }
+    }
+}
+```
+which is deployed:
+```bash
+$ dcos kafka-zookeeper --name=<service name> update start --options=kerberos-toggle-step-3.json
+$ dcos kafka-zookeeper --name=<service name> plan show deploy
+deploy (serial strategy) (COMPLETE)
+└─ node-update (serial strategy) (COMPLETE)
+   ├─ zookeeper-0:[server, metrics] (COMPLETE)
+   ├─ zookeeper-1:[server, metrics] (COMPLETE)
+   └─ zookeeper-2:[server, metrics] (COMPLETE)
+```
+
+Unauthenticated clients will now only be allowed to ping, create a session, close a session, or authenticate when communicating with the {{ model.techName }} instance.
+
+*Note*: The default settings for `service.security.kerberos.advanced.required_for_quorum_learner`, `service.security.kerberos.advanced.required_for_quorum_server`, `service.security.kerberos.advanced.required_for_client` are all `true`.
+
+#### Disabling Kerberos After Deployment
+
+*Note*: Disabling Kerberos after deployment is **not** supported.
+
+## Securely Exposing DC/OS {{ model.techName }} Outside the Cluster.
+
+Kerberos security is tightly coupled to the DNS hosts of the zookeeper tasks. As such, exposing a secure {{ model.techName }} service outside of the cluster requires additional setup.
+
+### Server to Client Connection
+
+To expose a secure {{ model.techName }} service outside of the cluster, any client connecting to it must be able to access all tasks of the service via the IP address assigned to the task. This IP address will be one of: an IP address on a virtual network or the IP address of the agent the task is running on.
```

Comment:
Note that we state that this cannot be run on a virtual network as a limitation. I assume that this paragraph indicates that the limitation is incorrect -- or this paragraph may need to be updated.